packages/arb-token-bridge-ui/src/components/TransferPanel/TokenRow.tsx

```diff
@@ -62,32 +62,86 @@ function BlockExplorerTokenLink({
   return (
     <ExternalLink
       href={`${getExplorerUrl(chain.id)}/token/${address}`}
-      className="text-xs text-blue-link underline"
+      className="arb-hover text-xs underline"
       onClick={e => e.stopPropagation()}
     >
       {shortenAddress(address).toLowerCase()}
     </ExternalLink>
   )
 }
 
+function TokenListInfo({ token }: { token: ERC20BridgeToken | null }) {
+  const [networks] = useNetworks()
+  const { childChain, childChainProvider } = useNetworksRelationship(networks)
+  const { isCustom: childChainNativeCurrencyIsCustom } = useNativeCurrency({
+    provider: childChainProvider
+  })
+
+  const tokenListInfo = useMemo(() => {
+    if (!token) {
+      return null
+    }
+
+    if (isTokenArbitrumOneNativeUSDC(token?.address)) {
+      return 'Native USDC on Arbitrum One'
+    }
+
+    if (isTokenArbitrumSepoliaNativeUSDC(token?.address)) {
+      return 'Native USDC on Arbitrum Sepolia'
+    }
+
+    const listIds: Set<number> = token.listIds
+    const listIdsSize = listIds.size
+    if (listIdsSize === 0) {
+      return 'Added by User'
+    }
+
+    const listIdsArray = Array.from(listIds)
+    if (listIdsSize < 2) {
+      return tokenListIdsToNames(listIdsArray)
+    }
+
+    const firstList = listIdsArray.slice(0, 1)
+    const more = listIdsSize - 1
+
+    return (
+      tokenListIdsToNames(firstList) +
+      ` and ${more} more list${more > 1 ? 's' : ''}`
+    )
+  }, [token])
+
+  if (!token) {
```

Comment:
Will this ever be reached? The same condition is at the very top

Reply:
yes, `token` can be `null` when passed into `TokenListInfo`
This is to also handle native gas token cases

Reply:
"the top" you saw is within `tokenListInfo`

Reply:
I see, my bad